### PR TITLE
Notifications send automatically #155

### LINKED ISSUE #155: Get notifications sending automatically
- Observer superclass to hold notifications
- asynch function that calls notify on interval in flask.

### PR DESCRIPTION
**Pull request by RyanPybus**
pr to close #155 
- On initialization sets up a function to run every 5 mins,
- this pulls events from the api from the last 5 mins, 
- compares against all notifications and sends emails for matches.
- requires Flask-APScheduler (added to requirements.txt)

- combines emails into one if multiple events trigger the same notification
image

### CODE REVIEW

File: flaskr/__init__.py [hunk 1]
```diff
@@ -11,6 +13,17 @@ def create_app(test_config=None):
         DATABASE=os.path.join(app.instance_path, 'flaskr.sqlite'),
     )
 
+    # initialize scheduler. Check for sending notifications every 5 mins (300 seconds)
+    scheduler = APScheduler()
```
Comment (anthonyargatoff) [inline flaskr/__init__.py:17]: Nice, this looks cool! I didn't know there was a scheduler in python!

**Review by anthonyargatoff — APPROVED** [review]
Notifications look great. Well done!

**Review by Varun2711 — APPROVED** [review]
Great work Ryan!!!

**Review by Aayushchaudhary84 — APPROVED** [review]
Looks Good

**Review by DyPickard — APPROVED** [review]
Yup looks good! I see no issues.

**Pull request merged**